### PR TITLE
Add YOLO-World model documentation

### PR DESCRIPTION
**Pull request by capjamesg**
# Description

This PR adds instructions on how to use YOLO-World to the Inference documentation.

## Type of change

- [ ] Documentation improvements

## How has this change been tested, please provide a testcase or example of how you tested the change?

You can test this project by navigating to the Models tab, click "YOLO World", and ensure the page opens properly.

## Any specific deployment considerations

N/A

## Docs

N/A

### CODE REVIEW

**Review by PawelPeczek-Roboflow — CHANGES_REQUESTED** [review]
We shall also include examples of running the requests against HTTP API with inference_sdk

**Comment by capjamesg** [comment]
@PawelPeczek-Roboflow Do you have code to share for those methods?

**Comment by PawelPeczek-Roboflow** [comment]
```python
result = client.infer_from_yolo_world(
    inference_input=["https://media.istockphoto.com/id/1417882544/photo/large-group-of-cats-and-dogs-looking-at-the-camera-on-blue-background.jpg?s=612x612&w=0&k=20&c=kGKANSIFdNfhBJMipyuaKU4BcVE1oELWev9lF2ickE0="]*3,
    class_names=["dog", "cat"],
    model_version="m",
    confidence=0.1,
)
```
there is also async version

```python
result = await client.infer_from_yolo_world_async(
    inference_input=["https://media.istockphoto.com/id/1417882544/photo/large-group-of-cats-and-dogs-looking-at-the-camera-on-blue-background.jpg?s=612x612&w=0&k=20&c=kGKANSIFdNfhBJMipyuaKU4BcVE1oELWev9lF2ickE0="]*3,
    class_names=["dog", "cat"],
    model_version="m",
    confidence=0.1,
)
```

and if one `inefrence server start` and `pip install inference-sdk` first then this should work out of the box

**Comment by PawelPeczek-Roboflow** [comment]
this is example for batch requests

**Comment by PawelPeczek-Roboflow** [comment]
one may also give np array, PIL or single file path / url - as per standard way of specifying images in sdk

File: docs/foundation/yolo_world.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    )
+
+    result = client.infer_from_yolo_world(
+        inference_input=["https://media.roboflow.com/dog.jpeg"],
+        class_names=["person", "backpack", "dog", "eye", "nose", "ear", "tongue"],
+        model_version="l",
+        confidence=0.1,
+    )
+
+    detections = sv.Detections.from_inference(results)
```
Comment (PawelPeczek-Roboflow) [inline docs/foundation/yolo_world.md:68]: afaik we return list of dictionaries, so probably results[0] is appropriate here

File: docs/foundation/yolo_world.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+=== "Inference Server API ()"
+
+    ```python
+    import cv2
+    import supervision as sv
+
+    from inference_sdk import InferenceHTTPClient
+
+    client = InferenceHTTPClient(
+        api_url="http://127.0.0.1:9001",
```
Comment (PawelPeczek-Roboflow) [inline docs/foundation/yolo_world.md:57]: we shall probably instruct users to run server with inference-cli and probably give a hint about `pip install ...`

**Comment by capjamesg** [comment]
I have made the requested changes. Can you review again? (I can't merge without the last commit being reviewed and approved)

**Review by PawelPeczek-Roboflow — APPROVED** [review]

**Pull request merged**